spring-xd-dirt/src/main/java/org/springframework/xd/dirt/zookeeper/ZooKeeperConnection.java

```diff
@@ -118,9 +150,29 @@ public ZooKeeperConnection(String clientConnectString) {
 	 * @param namespace the root path namespace in ZooKeeper (or default namespace if null)
 	 */
 	public ZooKeeperConnection(String clientConnectString, String namespace) {
+		this(clientConnectString, namespace, DEFAULT_SESSION_TIMEOUT, DEFAULT_CONNECTION_TIMEOUT,
+				DEFAULT_INITIAL_RETRY_WAIT, DEFAULT_MAX_RETRY_ATTEMPTS);
+	}
+
+	/**
+	 * Establish a ZooKeeper connection with the provided client connect string, namespace,
+	 * and timing values.
+	 *
+	 * @param clientConnectString one or more {@code host:port} strings, comma-delimited if more than one
+	 * @param namespace the root path namespace in ZooKeeper (or default namespace if null)
+	 * @param sessionTimeout ZooKeeper session timeout in milliseconds
+	 * @param connectionTimeout ZooKeeper connection timeout in milliseconds
+	 * @param retryWait milliseconds between connection retries
+	 * @param retryMaxAttempts number of connection attempts that will be made after a failed connection attempt
+	 */
+	public ZooKeeperConnection(String clientConnectString, String namespace, int sessionTimeout,
+			int connectionTimeout, int retryWait, int retryMaxAttempts) {
```

Comment:
For consistency, perhaps the "retryWait" arg name should be "initialRetryWait". Likewise, it would be good to provide a simple statement in the JavaDoc for this constructor that mentions it will use ExponentialBackoffRetry and include a "see" annotation so people can refer to that Curator class directly.